ui/src/app/pages/workspace/workspace-edit.spec.tsx

```diff
@@ -1100,138 +1015,123 @@ describe('WorkspaceEdit', () => {
-      'billing-account',
-    ]);
-    // @ts-ignore
-    expect(billingDropDown.props().options.map((o) => o.label)).toEqual([
-      'User Provided Billing Account',
-    ]);
-
-    // Now select SELECT_OR_CREATE_BILLING_ACCOUNT_OPTION_VALUE, expect request billing scope then show the
-    // real billing accounts user has access to.
-    mockHasBillingScope.mockImplementation(() => true);
-    wrapper
-      .find('[data-test-id="billing-dropdown-div"]')
-      .first()
-      .simulate('click');
-    await waitOneTickAndUpdate(wrapper);
+    expect(
+      await screen.findByDisplayValue(USER_PROVIDED_BILLING_OPTION_REGEX)
+    ).toBeInTheDocument();
 
-    billingDropDown = wrapper.find('[data-test-id="billing-dropdown"]').first();
+    // Open dropdown, this triggers updating billing options
+    await user.click(screen.getByTestId('billing-dropdown'));
     expect(mockEnsureBillingScope).toHaveBeenCalledTimes(1);
-    expect(billingDropDown.props().value).toEqual('billing-account');
-    // @ts-ignore
-    expect(billingDropDown.props().options.map((o) => o.value)).toEqual([
-      'free-tier',
-      'user-billing',
-      'billing-account',
-    ]);
-    // @ts-ignore
-    expect(billingDropDown.props().options.map((o) => o.label)).toEqual([
-      'Use All of Us initial credits - $33.33 left',
-      'User Billing',
-      'User Provided Billing Account',
-    ]);
+
+    // Close the dropdown in order to allow the dropdown to update?
+    await user.click(screen.getByTestId('billing-dropdown'));
+
+    expect(
+      screen.getAllByRole('option', {
+        name: FREE_TIER_OPTION_REGEX,
+        hidden: true,
+      })[0]
+    ).toBeInTheDocument();
+
+    expect(
+      screen.queryByRole('option', {
+        name: USER_BILLING_OPTION_REGEX,
+        hidden: true,
+      })
+    ).toBeInTheDocument();
+
+    expect(
+      screen.getAllByRole('option', {
+        name: USER_PROVIDED_BILLING_OPTION_REGEX,
+        hidden: true,
+      })[0]
+    ).toBeInTheDocument();
   });
   it('Should trim workspace name', async () => {
-    const wrapper = component();
-    let workspaceNameTextBox = wrapper
-      .find('[data-test-id="workspace-name"]')
-      .first();
-    workspaceNameTextBox.simulate('change', {
-      target: { value: '  ' + WORKSPACE_NAME_TEXT + '    ' },
-    });
-    expect(workspaceNameTextBox.prop('value')).not.toBe(WORKSPACE_NAME_TEXT);
-    workspaceNameTextBox.simulate('blur');
-    workspaceNameTextBox = wrapper
-      .find('[data-test-id="workspace-name"]')
-      .first();
-    expect(workspaceNameTextBox.prop('value')).toBe(WORKSPACE_NAME_TEXT);
+    renderComponent();
+    const workspaceNameTextBox = await screen.findByTestId('workspace-name');
+    await userEvent.type(
+      workspaceNameTextBox,
+      '  ' + WORKSPACE_NAME_TEXT + '    '
+    );
+    expect(
+      screen.queryByDisplayValue(WORKSPACE_NAME_TEXT, {
+        normalizer: getDefaultNormalizer({ trim: false }),
```

Comment:
what does this do?

Reply:
Good question! I added a comment for posterity, but by default, RTL trims leading and trailing whitespace from text it encounters in the DOM. So if the DOM reads:
<div>[SPACE][SPACE]Workspace Name[SPACE][SPACE]</div>
screen.getByText('Workspace Name'), which normally requires an exact match will match against the DOM above, because RTL is normalizing it to:
<div>Workspace Name</div>

In the case of this test, we want to verify when our application is applying whitespace trimming to the displayed text, and that would be hard to do when RTL is doing its own trimming. By adding `{normalizer: getDefaultNormalizer({ trim: false })` to the end of our query, we are preventing RTL from doing its own whitespace trimming.

More details can be found here:
https://testing-library.com/docs/queries/about/#normalization

Reply:
interesting!  thanks